dds/idl/marshal_generator.cpp

```diff
@@ -153,36 +153,29 @@ namespace {
 } /* namespace */
 
 bool marshal_generator::gen_enum(AST_Enum*, UTL_ScopedName* name,
-  const std::vector<AST_EnumVal*>& vals, const char*)
+  const std::vector<AST_EnumVal*>&, const char*)
 {
   NamespaceGuard ng;
   be_global->add_include("dds/DCPS/Serializer.h");
-  string cxx = scoped(name); // name as a C++ class
+  const string cxx = scoped(name), // name as a C++ class
+    underscores = scoped_helper(name, "_");
   {
     Function insertion("operator<<", "bool");
     insertion.addArg("strm", "Serializer&");
     insertion.addArg("enumval", "const " + cxx + "&");
     insertion.endArgs();
-    const std::string idl_name = canonical_name(name);
     be_global->impl_ <<
-      "    if (CORBA::ULong(enumval) >= " << vals.size() << ") {\n"
-      "      if (OpenDDS::DCPS::log_level >= OpenDDS::DCPS::LogLevel::Warning) {\n"
-      "        ACE_ERROR((LM_WARNING, \"(%P|%t) WARNING: "
-        "%u is an invalid enumerated value for " << idl_name << "\\n\", enumval));\n"
-      "      }\n"
-      "      return false;\n"
-      "    }\n"
-      "  return strm << static_cast<CORBA::ULong>(enumval);\n";
```

Comment:
Is it okay to take out this warning? It seems reasonable. Is the check for validity really that costly?

Reply:
Since this is the output side, it's not necessary to check.  But checking may be helpful/desirable.
Currently it's just a `O(N)` walk through the list of valid options.  For some enums this might be noticeable since it's on the ciritical path.